crates/hyperdrive-math/src/lp.rs

```diff
@@ -600,6 +621,540 @@ impl State {
+                    // recalculate the max bond amount.
+                    //
+                    // NOTE: We are calling this with 'self' so that original
+                    // values are used with the updated value of share_proceeds.
+                    let updated_state =
+                        self.get_state_after_liquidity_update(-I256::try_from(share_proceeds)?)?;
+                    let max_bond_amount = updated_state
+                        .calculate_max_sell_bonds_in_safe(self.minimum_share_reserves())?;
+
+                    // If the max bond amount is less than or equal to the net
+                    // curve trade, then Newton's method has terminated since
+                    // proceeding to the next step would result in reaching the
+                    // same point.
+                    if max_bond_amount <= net_curve_trade {
+                        return Ok(share_proceeds);
+                    }
+                    // Otherwise, we continue to the next iteration of Newton's
+                    // method.
+                }
+            }
+
+            // We calculate the derivative of F(x) using the derivative of
+            // `calculateSharesOutGivenBondsIn` when the pool is net long or
+            // the derivative of `calculateSharesInGivenBondsOut`. when the pool
+            // is net short.
+            let derivative = updated_state.calculate_shares_delta_given_bonds_delta_derivative(
+                net_curve_trade,
+                self.share_reserves(),
+                self.bond_reserves(),
+                self.effective_share_reserves(),
+                self.share_adjustment(),
+            )?;
+            if derivative >= fixed!(1e18) {
+                return Err(eyre!("Derivative is greater than one"));
+            }
+            let derivative = fixed!(1e18) - derivative;
+
+            // NOTE: Round the delta down to avoid overshooting.
+            //
+            // Calculate the objective function's value. If the value's magnitude
+            // is smaller than the previous smallest value, then we update the
+            // value and record the share proceeds. We'll ultimately return the
+            // share proceeds that resulted in the smallest value.
+            let delta = I256::try_from(present_value.mul_down(lp_total_supply))?
+                - I256::try_from(starting_present_value.mul_up(active_lp_total_supply))?;
+            if smallest_delta == I256::from(0) || delta.abs() < smallest_delta.abs() {
+                smallest_delta = delta;
+                closest_share_proceeds = share_proceeds;
+                closest_present_value = present_value;
+            }
+
+            // We calculate the updated share proceeds `x_n+1` by proceeding
+            // with Newton's method. This is given by:
+            //
+            // x_n+1 = x_n - F(x_n) / F'(x_n)
+            match delta.cmp(&I256::zero()) {
```

Comment:
I haven't had a chance to get around to this, but we don't like this syntax and will be ignoring the lint error. I don't think it is required to pass CI anyway. If possible, please change to the usual syntax and if not please add a TODO to change it back so we can find it later.

https://github.com/delvtech/hyperdrive-rs/issues/66